### PR TITLE
update entrypoint to work with rootless docker and be more conservative

### PR DESCRIPTION
**Pull request by ftalbrecht**

### CODE REVIEW

File: debian/common_context/entrypoint.sh [hunk 1]
```diff
@@ -9,27 +9,44 @@
...
-groupadd -g $GID_ $USERNAME_ &> /dev/null
-if [ -e /home/$USERNAME_ ] ; then
-  useradd -d /home/$USERNAME_ -g $GID_ -s /bin/bash -u $UID_ $USERNAME_
-else
-  useradd -md /home/$USERNAME_ -g $GID_ -s /bin/bash -u $UID_ $USERNAME_
-fi
+export LANG=en_US.UTF-8
+echo "127.0.0.1 ${HOSTNAME}" >> /etc/hosts
 
-export ALLINEA_LICENCE_FILE=/home/$USERNAME_/.config/allinea/Licence \
```
Comment (renefritze) [inline debian/common_context/entrypoint.sh:23]: Where did this go?

File: debian/common_context/entrypoint.sh [hunk 2]
```diff
@@ -9,27 +9,44 @@
...
+else
+  # otherwise create the user
+  [[ -e /home/$USERNAME_ ]] && export m_= || export m_=m
+  useradd -${m_}d /home/$USERNAME_ -g $GID_ -s /bin/bash -u $UID_ $USERNAME_
+fi
 
-export LANG=en_US.UTF-8
-echo "127.0.0.1 ${HOSTNAME}" >> /etc/hosts
+# give the user some sudo capabilities
+if [[ $UID_ != 0 ]] ; then
```
Comment (renefritze) [inline debian/common_context/entrypoint.sh:45]: What's the benefit of guarding these?

File: debian/common_context/entrypoint.sh [hunk 3]
```diff
@@ -9,27 +9,44 @@
...
 GID_=${LOCAL_GID:-$UID_}
 
-groupadd -g $GID_ $USERNAME_ &> /dev/null
-if [ -e /home/$USERNAME_ ] ; then
-  useradd -d /home/$USERNAME_ -g $GID_ -s /bin/bash -u $UID_ $USERNAME_
-else
-  useradd -md /home/$USERNAME_ -g $GID_ -s /bin/bash -u $UID_ $USERNAME_
-fi
+export LANG=en_US.UTF-8
+echo "127.0.0.1 ${HOSTNAME}" >> /etc/hosts
```
Comment (renefritze) [inline debian/common_context/entrypoint.sh:17]: Why is this necessary?

File: debian/common_context/entrypoint.sh [hunk 4]
```diff
@@ -9,27 +9,44 @@
...
+UID_=${LOCAL_UID:-0}
 GID_=${LOCAL_GID:-$UID_}
 
-groupadd -g $GID_ $USERNAME_ &> /dev/null
-if [ -e /home/$USERNAME_ ] ; then
-  useradd -d /home/$USERNAME_ -g $GID_ -s /bin/bash -u $UID_ $USERNAME_
-else
-  useradd -md /home/$USERNAME_ -g $GID_ -s /bin/bash -u $UID_ $USERNAME_
-fi
+export LANG=en_US.UTF-8
```
Comment (renefritze) [inline debian/common_context/entrypoint.sh:16]: This should already be baked into the image. If would be better not to repeat it then.

File: debian/common_context/entrypoint.sh (hunk 1 above)
Comment (ftalbrecht) [reply debian/common_context/entrypoint.sh:23]: Dropped it, sorry. Is the trailing `\` on purpose?

File: debian/common_context/entrypoint.sh (hunk 2 above)
Comment (ftalbrecht) [reply debian/common_context/entrypoint.sh:45]: I just found it unnatural to add root to the sudoers file, objections?

File: debian/common_context/entrypoint.sh (hunk 3 above)
Comment (ftalbrecht) [reply debian/common_context/entrypoint.sh:17]: Actually, I do not recall. It was present in the entrypoint before so I just kept it.

File: debian/common_context/entrypoint.sh (hunk 1 above)
Comment (renefritze) [reply debian/common_context/entrypoint.sh:23]: I guess in the original there was more than one var exported?

File: debian/common_context/entrypoint.sh (hunk 4 above)
Comment (ftalbrecht) [reply debian/common_context/entrypoint.sh:16]: agreed. similarly the hosts definition, I guess?

File: debian/common_context/entrypoint.sh (hunk 2 above)
Comment (renefritze) [reply debian/common_context/entrypoint.sh:45]: No, just found it weird.

File: debian/common_context/entrypoint.sh (hunk 3 above)
Comment (renefritze) [reply debian/common_context/entrypoint.sh:17]: Ah, further down in the diff, it moved up. Maybe this is useful if one sets the hostname at `docker run` time. DIdn't think of that earlier.

File: debian/common_context/entrypoint.sh (hunk 4 above)
Comment (renefritze) [reply debian/common_context/entrypoint.sh:16]: Actually, probably no (see above)?

File: debian/common_context/entrypoint.sh (hunk 3 above)
Comment (ftalbrecht) [reply debian/common_context/entrypoint.sh:17]: Yup, our docker_run.sh sets the hostname, so it fits well I think

File: debian/common_context/entrypoint.sh (hunk 4 above)
Comment (ftalbrecht) [reply debian/common_context/entrypoint.sh:16]: k, will prepare another pr. will also try to adapt the docker_run accordingly.

**Comment by tobiasleibner** [comment]
This seems to have broken ``ccache`` in ``dune-xt``. [This pipeline](https://zivgitlab.uni-muenster.de/ag-ohlberger/dune-community/dune-xt/-/pipelines/65874) still uses, e.g.,  ``/usr/lib/ccache/g++``, [the newest pipeline](https://zivgitlab.uni-muenster.de/ag-ohlberger/dune-community/dune-xt/-/pipelines/66355) uses ``/usr/bin/g++`` and the ``ccache`` command fails with ``/bin/bash: line 130: ccache: command not found``. I am not sure why, but this PR is my main suspect.

**Comment by ftalbrecht** [comment]
Ok. I will revert this and prepare another pr with the additional changes we discussed above.

**Comment by tobiasleibner** [comment]
Thanks.
